drivers/sx127x/sx127x_netdev.c

```diff
@@ -0,0 +1,513 @@
+                    /* Invert and divide by 4 */
+                    packet_info->snr = -1 * ((~snr_value + 1) & 0xFF) >> 2;
+                }
+                else {
+                    /* Divide by 4 */
+                    packet_info->snr = (snr_value & 0xFF) >> 2;
+                }
+
+                int16_t rssi = sx127x_reg_read(dev, SX127X_REG_LR_PKTRSSIVALUE);
+
+                if (packet_info->snr < 0) {
+#if defined(MODULE_SX1272)
+                    packet_info->rssi = SX127X_RSSI_OFFSET + rssi + (rssi >> 4) + packet_info->snr;
+#else /* MODULE_SX1276 */
+                    if (dev->settings.channel > SX127X_RF_MID_BAND_THRESH) {
+                        packet_info->rssi = SX127X_RSSI_OFFSET_HF + rssi + (rssi >> 4) + packet_info->snr;
+                    }
+                    else {
+                        packet_info->rssi = SX127X_RSSI_OFFSET_LF + rssi + (rssi >> 4) + packet_info->snr;
+                    }
+#endif
+                }
+                else {
+#if defined(MODULE_SX1272)
+                    packet_info->rssi = SX127X_RSSI_OFFSET + rssi + (rssi >> 4);
+#else /* MODULE_SX1276 */
+                    if (dev->settings.channel > SX127X_RF_MID_BAND_THRESH) {
+                        packet_info->rssi = SX127X_RSSI_OFFSET_HF + rssi + (rssi >> 4);
+                    }
+                    else {
+                        packet_info->rssi = SX127X_RSSI_OFFSET_LF + rssi + (rssi >> 4);
+                    }
+#endif
+                }
+            }
+
+            size = sx127x_reg_read(dev, SX127X_REG_LR_RXNBBYTES);
+            if (buf == NULL) {
+                return size;
+            }
+
+            if (size > len) {
+                return -ENOBUFS;
+            }
+
+            if (!dev->settings.lora.rx_continuous) {
+                sx127x_set_state(dev, SX127X_RF_IDLE);
+            }
+
+            xtimer_remove(&dev->_internal.rx_timeout_timer);
+
+            /* Read the last packet from FIFO */
+            uint8_t last_rx_addr = sx127x_reg_read(dev, SX127X_REG_LR_FIFORXCURRENTADDR);
+            sx127x_reg_write(dev, SX127X_REG_LR_FIFOADDRPTR, last_rx_addr);
+            sx127x_read_fifo(dev, (uint8_t*)buf, size);
+            break;
+        default:
+            break;
+    }
+
+    return size;
+}
+
+static int _init(netdev_t *netdev)
+{
+    sx127x_t *sx127x = (sx127x_t*) netdev;
+
+    sx127x->irq = 0;
+    sx127x_radio_settings_t settings;
+    settings.channel = SX127X_CHANNEL_DEFAULT;
+    settings.modem = SX127X_MODEM_DEFAULT;
+    settings.state = SX127X_RF_IDLE;
+
+    sx127x->settings = settings;
+
+    /* Launch initialization of driver and device */
+    DEBUG("init_radio: initializing driver...\n");
+    sx127x_init(sx127x);
+
+    sx127x_init_radio_settings(sx127x);
+    /* Put chip into sleep */
+    sx127x_set_sleep(sx127x);
+
+    DEBUG("init_radio: sx127x initialization done\n");
+
+    return 0;
+}
+
+static void _isr(netdev_t *netdev)
```

Comment:
There 6 irq pins and first 4 are actually used for LoRa and FSK. The 2 remaining ones are FSK specific. The actual version of this driver doesn't support FSK